### PR TITLE
TET-758: Only replace history if params has values

### PR DESCRIPTION
**Pull request by baarkerlounger**
## Description of change

Prior to V5 ` history.replace(url, { data: res.data })` used to strip trailing `?`s. It now doesn't causing a test to fail, this may be related to https://github.com/remix-run/history/issues/701.

This PR checks if we actually have any data in the params object before calling `history.replace()` at all, since that seems in like with the test intentions.

Alternative approaches:
We could adjust the test expectations (there's not much harm in leaving the trailing `?`), or we could strip trailing `?`s from `url` before passing it into `history.replace`.

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'

- [ ] Has the branch been rebased to main?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

### CODE REVIEW

**Review by santoshdasa12345 — APPROVED** [review]

File: test/unit-client/assets/javascripts/lib/xhr.test.js [hunk 1]
```diff
@@ -40,6 +40,8 @@ describe('XHR', () => {
       const res = { data: {} }
       const params = {}
       XHR.updateOutlet(res, params)
+      if (Object.keys(params).length > 1) {
+      }
```
Comment (marijnkampf) [inline test/unit-client/assets/javascripts/lib/xhr.test.js:44]: Either there is a expect / assert missing, or these two lines are not needed.

File: assets/javascripts/lib/xhr.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ const XHR = {
   updateOutlet(res, params, win = window) {
     this.injectResponseInHtml(res.data)
 
-    if (params) {
+    if (params && Object.keys(params).length > 1) {
```
Comment (baarkerlounger) [inline assets/javascripts/lib/xhr.js:42]: This should probably be `> 0` actually?

File: test/unit-client/assets/javascripts/lib/xhr.test.js [hunk 3]
```diff
@@ -40,6 +40,8 @@ describe('XHR', () => {
       const res = { data: {} }
       const params = {}
       XHR.updateOutlet(res, params)
+      if (Object.keys(params).length > 1) {
```
Comment (baarkerlounger) [inline test/unit-client/assets/javascripts/lib/xhr.test.js:43]: What is this supposed to be doing?

File: assets/javascripts/lib/xhr.js (hunk 2 above)
Comment (santoshdasa12345) [reply assets/javascripts/lib/xhr.js]: resolved

File: test/unit-client/assets/javascripts/lib/xhr.test.js (hunk 1 above)
Comment (santoshdasa12345) [reply test/unit-client/assets/javascripts/lib/xhr.test.js]: Added assertion..resolved

File: test/unit-client/assets/javascripts/lib/xhr.test.js [hunk 4]
```diff
@@ -40,6 +40,7 @@ describe('XHR', () => {
       const res = { data: {} }
       const params = {}
       XHR.updateOutlet(res, params)
+      expect(Object.keys(params).length).to.equal(0)
```
Comment (baarkerlounger) [inline test/unit-client/assets/javascripts/lib/xhr.test.js:43]: What is this assertion supposed to be testing?

**Review by swenban — APPROVED** [review]

**Pull request merged**